automated install/basic-install.sh

```diff
@@ -353,7 +415,135 @@ build_dependency_package(){
+            installed_files+=("user:builder")
+        fi
+
+        # Create necessary directories and prepare build environment
+        printf "  %b Setting up build environment..." "${INFO}"
+        mkdir -p /var/lib/builder/pihole-meta
+        mkdir -p /var/lib/builder/.abuild
+        mkdir -p /etc/pihole
+        chown -R builder:abuild /var/lib/builder
+        # Track Alpine-specific changes
+        touch /etc/pihole/alpine_installed_files
+        touch /etc/pihole/alpine_modified_files
+        printf "%b  %b Done\\n" "${OVER}" "${TICK}"
+
+        # Generate abuild keys for builder user if they don't exist
+        printf "  %b Generating abuild keys..." "${INFO}"
+
+        # Generate keys and capture the output
+        # builder does not trust keys if created under a permissive umask
+        original_umask="$(umask)"
+        umask 0002
+        key_output=$(su builder -c "cd /var/lib/builder/.abuild && abuild-keygen -n" 2>&1)
+        if [ $? -ne 0 ]; then
+            printf "%b  %b Failed to generate keys\\n" "${OVER}" "${CROSS}"
+            return_code=1
+            return ${return_code}
+        fi
+
+        # Extract the private key path from the output
+        private_key=$(echo "$key_output" | grep "PACKAGER_PRIVKEY=" | cut -d'"' -f2)
+        if [ -n "$private_key" ] && [ -f "$private_key" ]; then
+            # Update abuild.conf with the exact key path from output
+            cat > /var/lib/builder/.abuild/abuild.conf << EOF
+PACKAGER_PRIVKEY="$private_key"
+PACKAGER="builder <builder@localhost>"
+EOF
+            chmod 600 /var/lib/builder/.abuild/abuild.conf
+            chown -R builder:abuild /var/lib/builder/.abuild
+
+            # Copy public key to system keys
+            public_key="${private_key}.pub"
+            if [ -f "$public_key" ]; then
+                mkdir -p /etc/apk/keys
+                cp "$public_key" /etc/apk/keys/
+                printf "%b  %b Done\\n" "${OVER}" "${TICK}"
+            else
+                printf "%b  %b Failed to find public key\\n" "${OVER}" "${CROSS}"
+                return_code=1
+                return ${return_code}
+            fi
+        else
+            printf "%b  %b Failed to find private key\\n" "${OVER}" "${CROSS}"
+            return_code=1
+            return ${return_code}
+        fi
+
+        # Write the APKBUILD file
+        printf "  %b Creating APKBUILD..." "${INFO}"
+        echo "${PIHOLE_META_PACKAGE_CONTROL_APK}" > /var/lib/builder/pihole-meta/APKBUILD
+        chown -R builder:abuild /var/lib/builder/pihole-meta
+        printf "%b  %b Done\\n" "${OVER}" "${TICK}"
+
+        # Install build dependencies
+        printf "  %b Installing build dependencies..." "${INFO}"
+        if ! apk add --no-cache build-base bind-tools binutils coreutils curl git grep iproute2-ss jq libcap logrotate ncurses nmap-ncat procps-ng psmisc shadow tzdata unzip wget >/dev/null 2>&1; then
```

Comment:
Do we really need all those packages as build dependencies?

Reply:
I believe so. It seems comparable in number to the other package managers.

Reply:
Ah yes. I see now.
I will have a look at thinning those out.